code/game/objects/structures/popout_cake.dm

```diff
@@ -0,0 +1,185 @@
+	max_integrity = 100
+	integrity_failure = 10
+	move_resist = MOVE_FORCE_WEAK
+	layer = OBJ_LAYER
+	var/mob/living/occupant = null
+	var/datum/action/item_action/pull_string/string //Action for pulling the string for a surprise reveal
+	var/used_string = FALSE //If the string for a surprise reveal has been pulled from inside
+	var/amount_of_slices = 16 //How many cake slices will appear once it's cut up
+	var/slice_path  = /obj/item/reagent_containers/food/snacks/cakeslice/plain/full //What kind of cake slice will appear
+	var/strong_surprise = FALSE //If the surprise reveal has an extra oomph to it, used for the newcop exclusive cake
+
+/obj/structure/popout_cake/newcop
+	strong_surprise = TRUE
+
+/obj/structure/popout_cake/Initialize(mapload)
+	. = ..()
+	if(!string)
+		var/datum/action/item_action/pull_string/S = new()
+		string = S
+		string.cake = src
+
+/obj/structure/popout_cake/MouseDrop_T(mob/living/target, mob/user)
+	if(!istype(target) || user.incapacitated() || !target.Adjacent(user) || !Adjacent(user) || !ismob(target) || (!ishuman(user) && !iscyborg(user)) || !istype(user.loc, /turf) || target.buckled)
+		return
+
+	if(occupant)
+		to_chat(user, "<span class='warning'>There's already someone inside!</span>")
+		return
+
+	if(obj_integrity <= integrity_failure)
+		to_chat(user, "<span class='warning'>The [src] is too damaged to hold anyone inside!</span>")
+		return
+
+	if(target == user)
+		user.visible_message("<span class = 'notice'>[user] starts climbing into [src].</span>", "<span class = 'notice'>You start climbing into [src].</span>")
+	else
+		user.visible_message("<span class = 'warning'>[user] starts stuffing [target] into [src]!</span>", "<span class = 'warning'>You start stuffing [target] into [src]!</span>")
+
+	if(do_after(user, 60, TRUE, src))
+		if(occupant)
+			to_chat(user, "<span class='warning'>There's already someone inside!</span>")
+			return
+		if(target != user)
+			to_chat(user, "<span class = 'notice'>You stuff [target] into [src]!</span>")
+		target.forceMove(src)
+		occupant = target
+		if(target != user)
+			log_combat(user, occupant, "stuffed ", null, "into [src]")
+		string.Grant(occupant)
+		to_chat(occupant, "<span class='notice'>You are now inside the cake! When you're ready to emerge from the cake in a blaze of confetti and party horns, \
+		pull on the string(<b>It will have to be wound back up with a screwdriver if you want to do it again</b>). If you wish to leave without setting off the confetti, just attempt to move out of the cake!</span>")
+	add_fingerprint(target)
+
+/obj/structure/popout_cake/relaymove(mob/user, direction)
+	if(!istype(user))
+		return
+	else if(contents.Find(user))
+		to_chat(user, "<span class='info'>You begin climbing out of the [src]...</span>")
+		if(do_after(user, 20))
+			user.forceMove(get_turf(src))
+			occupant = null
+
+/obj/structure/popout_cake/attackby(obj/item/W, mob/user, params)
+	if(W.tool_behaviour == TOOL_SCREWDRIVER && used_string == TRUE)
```

Comment:
can't this use screwdriver_act?

Reply:
I honestly don't know, screwdriver_act is usually used for opening the wire panel. Maybe it'll mess some things up, I'll need to test it.

Reply:
Tried it, doesn't work like I want it to.